### ISSUE TITLE
TypeError in Polygon.__del__: 'NoneType' object is not callable

### ISSUE THREAD

**Issue by privatwolke** [Toblerity/Shapely#473]
## Expected behavior and actual behavior.

Expected: Shapely disposes of resources cleanly and without raising exceptions.

Actual: Shapely does some kind of cleanup on the `Polygon` class which raises a TypeError. According to the Python documentation it's bad practice to mess with `__del__` at all. Exceptions raised during `__del__` are ignored by the interpreter and printed as warnings. This does not impact functionality but is really annoying and smells of potential memory leaks.

## Steps to reproduce the problem.

```
>>> from shapely.geometry.polygon import Polygon
>>> <Ctrl+D>
Exception TypeError: TypeError("'NoneType' object is not callable",) in <bound method Polygon.__del__ of <shapely.geometry.polygon.Polygon object at 0x109863810>> ignored
Exception TypeError: TypeError("'NoneType' object is not callable",) in <bound method Polygon.__del__ of <shapely.geometry.polygon.Polygon object at 0x10a3e80d0>> ignored
```

## Operating system

macOS Sierra 10.12.3 as well as on alpine linux 3.5; on python 2.7

## Shapely version and provenance

```
>>> from shapely import __version__
>>> __version__
'1.6b4'
```

Retrieved via PyPi at https://pypi.python.org/pypi/Shapely/1.6b4

**Comment by sgillies** [comment]
@privatwolke I believe you, but I can't replicate the above on my Macbook with Python 2.7.11 downloaded from python.org.

We've dealt with a possibly related issue in the past: https://github.com/Toblerity/Shapely/issues/264. Have you seen this problem with previous versions of Shapely?

**Comment by jtschoonhoven** [comment]
Fixed locally for me with https://github.com/Toblerity/Shapely/pull/528

**Comment by sgillies** [comment]
Closed via #528
